incubator/hnc/internal/reconcilers/anchor.go

```diff
@@ -72,12 +73,21 @@ func (r *AnchorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, err
 	}
 
-	// Report "Forbidden" state and early exit if the namespace is not allowed to have subnamespaces
-	// but has bypassed the webhook and successfully created the anchor. Forbidden anchors won't have
-	// finalizers.
-	// TODO refactor/split the ExcludedNamespaces map for 1) reconciler exclusion and 2) subnamespaces exclusion
-	//  purposes. See issue: https://github.com/kubernetes-sigs/multi-tenancy/issues/495
+	// Always delete anchor (and any other HNC CRs) in the excluded namespaces and
+	// early exit.
 	if config.ExcludedNamespaces[pnm] {
+		// Since the anchors in the excluded namespaces are never synced by HNC,
```

Comment:
Just to double check. In the scenario that we have an existing structure with multiple anchors, and the allowCascadingDelete is true, if we update HNC to now exclude the parent namespace, this will delete the anchor resources but NOT delete the children namespaces and they will be be parent-less, correct?

Reply:
Ah I didn't think of this scenario.

I just tested this scenario, both the subnamespace and anchor were not deleted (before I manually remove the finalizer in the anchor).

I created 2 level subns in ns `ex`, then exclude the `ex` and redeploy HNC. After that, I got:
```
$ k hns tree ex
ex
└── [s] sub (1,2)
    └── [s] grand (3)

[s] indicates subnamespaces

Conditions:
1) BadConfiguration (SubnamespaceAnchorMissing): The anchor is missing in the parent namespace
2) ActivitiesHalted (IllegalParent): Parent "ex" is an excluded namespace
3) ActivitiesHalted (AncestorHaltActivities): Propagation paused in the subtree of "sub" due to ActivitiesHalted condition

$ k get subns -A
NAMESPACE   NAME    AGE
ex          sub     12m
sub         grand   8m53s
```

The `badconfiguration` condition was due to excluded namespaces are not synced in the forest (memory), so it's invisible when we check forest.

The anchor object was still there even though we tried deleting it here. The reason is that after redeploying HNC, this anchor reconciliation always early exit here so the finalizers are not removed.

I then manually delete the anchor finalizer, the anchor was gone and the subnamespace `sub` and `grand` were still there having the same condition.

So everything looks like the intended behavior (conditions reported, not deleting ns), except my manual removal of anchor finalizers in the new excluded namespace. I will add this guidance in the excluded-namespace configuration doc. Does it sgty?

Reply:
Awesome yep sgtm.